pyi.py

```diff
@@ -170,16 +185,26 @@ def in_class(self) -> bool:
         return bool(self._class_nesting)
 
     def visit_Assign(self, node: ast.Assign) -> None:
-        self.generic_visit(node)
         if self.in_function:
             # We error for unexpected things within functions separately.
+            self.generic_visit(node)
             return
         if len(node.targets) != 1:
             self.error(node, Y017)
-            return
-        target = node.targets[0]
-        if not isinstance(target, ast.Name):
-            self.error(node, Y017)
+            target_name = None
+        else:
+            target = node.targets[0]
+            if not isinstance(target, ast.Name):
```

Comment:
Could we have the positive case first:

```python
if isinstance(target, ast.Name):
    bar
else:
    baz
```

instead of

```python
if not isinstance(target, ast.Name):
    baz
else:
    bar
```

Reply:
Done